### DISCUSSION TITLE
Using new PrimeVue versions with browser not supporting @layers

### DISCUSSION THREAD

**Discussion by micvm** [primefaces/community#884] [PrimeVue]
Since primevue introduced css layers, all browsers not supporting the @layer-feature cannot properly display PrimeVue applications due to broken css styles. For theme files, postcss can be used to convert the css for older browsers. But what about inline styles injected to the DOM by PrimeVue components, such as the Avatar component? What is the way to go in that case to achieve best compatibility?

**Comment by tugcekucukoglu** [comment]
We will re-work on @layer css. For now, this [theming section](https://primevue.org/theming/) and we can also recommend using [passthrough](https://primevue.org/passthrough/) implementation.

**Reply by micvm** [reply]
Thanks for the update (and all your work) about the layer-rework.

Do I understand it right: in terms of compatibility with older browsers (e.g. prior Chrome 97) the theming and passthrough is a possible solution for my own modifications. However, for the inline styles injected for certain Component by PrimeVue in the DOM it is less effort to wait for the re-work, right?

**Reply by egorshubin** [reply]
This layers feature breaks everything after an update. I'm using PrimeVue for a Prestashop module. And styles by Prestashop override my styles now! Because of the layers! So could you add an option to disable layers??

**Comment by cjbathras** [comment]
I am forced to support Chrome 79 by my customer so the @layer feature breaks everything and I have to pin to version 3.34.1. Would be great if we could toggle this feature depending on CSS Target. I'll also look into the theming and passthrough.

**Reply by egorshubin** [reply]
I also needed to update PrimeVue, but needed to remove layers at all. And I managed to do it.
I used these instructions to compile the theme from their SCSS theme. https://primevue.org/theming/#customtheme
I just removed @layer primevue from the downloaded SCSS files, so the theme is updated and doesn't have layers)

**Reply by micvm** [reply]
Thanks for sharing. At least for me this is not enough, since certain component styles are injected inline when using them. E.g. for the Menu component you can find the following within your DOM:
```html
<style type="text/css" data-primevue-style-id="menu">
@layer primevue {
    .p-menu ul {
        margin: 0;
        padding: 0;
        list-style: none;
    }

    .p-menu .p-menuitem-link {
        cursor: pointer;
        display: flex;
        align-items: center;
        text-decoration: none;
        overflow: hidden;
        position: relative;
    }

    .p-menu .p-menuitem-text {
        line-height: 1;
    }
}
</style>
```

**Reply by yarotskiu** [reply]
Hey @micvm, I am trying to find a way how to disable the @layer feature at the moment too. We need to support quite old browsers and I was looking into the same solution: copy a theme CSS file into the project and remove @layer manually. But to be sure that we are not using components with inline styles that are using the @layer feature, I cloned the PrimeVue repository and searched for @layer. And I have found 0 usage of @layer in components. Is that not the case anymore? 

Also, any updates on release date for:
<img width="540" alt="image" src="https://github.com/primefaces/community/assets/119170305/66fc601d-a89f-4ae2-afe3-bd327edac132">

From what I understand, this update will remove the '@layer' feature from CSS files, won't it?

**Reply by micvm** [reply]
I had a quick check on the code and found this commit: [60f08422e6](https://github.com/primefaces/primevue/commit/60f08422e68f17c797549f38292cf0d1a055cdb9) which should be part of release 3.48.0. It looks like they moved the inline styles away.

**Reply by yarotskiu** [reply]
Good to know! Thank you very much for checking the source code!
